src/executor_factory.py

```diff
@@ -185,35 +201,47 @@ def debug_executor(self, output_path, executor_name, num_approach, packages, des
             if error:
                 os.makedirs(next_executor_path)
                 file_name_to_content = self.get_all_executor_files_with_content(previous_executor_path)
-                all_files_string = self.files_to_string(file_name_to_content)
-                user_query = (
-                        f"General rules: " + not_allowed()
-                        + 'Here is the description of the task the executor must solve:\n'
-                        + description
-                        + '\n\nHere is the test scenario the executor must pass:\n'
-                        + test
-                        + 'Here are all the files I use:\n'
-                        + all_files_string
-                        + (('This is an error that is already fixed before:\n'
-                            + error_before) if error_before else '')
-                        + '\n\nNow, I get the following error:\n'
-                        + error + '\n'
-                        + 'Think quickly about possible reasons the error might caused by. '
-                          'Decide which files need to be changed. '
-                          'Then output the files that need change. '
-                          "Don't output files that don't need change. "
-                          "If you output a file, then write the complete file. "
-                          "Use the exact same syntax to wrap the code:\n"
-                          f"**...**\n"
-                          f"```...\n"
-                          f"...code...\n"
-                          f"```\n\n"
-                )
+
+                is_dependency_issue = self.is_dependency_issue(error, file_name_to_content['Dockerfile'])
+
+                if is_dependency_issue:
+                    all_files_string = self.files_to_string({
+                        key: val for key, val in file_name_to_content.items() if key in ['requirements.txt', 'Dockerfile']
+                    })
+                    user_query = (
+                        f"Your task is to provide guidance on how to solve an error that occurred during the Docker "
+                        f"build process. The error message is:\n{error}\nTo solve this error, you should first "
+                        f"identify the type of error by examining the stack trace. Once you have identified the "
+                        f"error, you should suggest how to solve it. Your response should include the files that "
+                        f"need to be changed, but not files that don't need to be changed. For files that need to "
+                        f"be changed, please provide the complete file with the exact same syntax to wrap the code.\n\n"
```

Comment:
drop please

Reply:
What was this about? I think file format? Not critical I guess.